src/nix-collect-garbage/nix-collect-garbage.cc

```diff
@@ -19,7 +19,10 @@ bool dryRun = false;
 
 void removeOldGenerations(std::string dir)
 {
-    if (access(dir.c_str(), R_OK) != 0) return;
+    if (access(dir.c_str(), R_OK) != 0) {
+        printError(format("warning: could not remove generations in: %1%") % dir);
```

Comment:
```suggestion
        printError(format("warning: cannot remove generation %1%: Permission denied") % dir);
```
Be more explicit about the type of error, follow the same pattern as `rm`
```
$ rm /etc/hosts 
rm: cannot remove '/etc/hosts': Permission denied
```

Reply:
Note that `format` is unnecessary. You can also do:
```
        printError("warning: cannot remove generation '%s': Permission denied", dir);
```